eras/shelley/impl/src/Cardano/Ledger/Shelley/TxWits.hs

```diff
@@ -249,23 +220,26 @@ shelleyEqTxWitsRaw txWits1 txWits2 =
     && liftEq eqRaw (txWits1 ^. scriptTxWitsL) (txWits2 ^. scriptTxWitsL)
     && liftEq eqRaw (txWits1 ^. bootAddrTxWitsL) (txWits2 ^. bootAddrTxWitsL)
 
-instance SafeToHash (ShelleyTxWits era) where
-  originalBytes (ShelleyTxWitsConstr w) = BSL.toStrict $ txWitsBytes w
-
 -- | Exports the relevant parts from a (WintessSetHKD Identity era) for
 --     use by the pretty printer without all the horrible constraints.
 --     Uses the non-exported WitnessSet' constructor.
 prettyWitnessSetParts ::
+  EraScript era =>
   ShelleyTxWits era ->
   ( Set (WitVKey 'Witness (EraCrypto era))
   , Map (ScriptHash (EraCrypto era)) (Script era)
   , Set (BootstrapWitness (EraCrypto era))
   )
-prettyWitnessSetParts (ShelleyTxWitsConstr (WitnessSet' a b c _)) = (a, b, c)
+prettyWitnessSetParts (ShelleyTxWits a b c) = (a, b, c)
```

Comment:
I think we can remove this function and just use the `ShelleyTxWits` in the place where it is being used in the tests